### PR TITLE
Upgrade to open-zeppelin-solidity

### PR DESCRIPTION
**Pull request by felix2feng**

### CODE REVIEW

File: contracts/core/SetToken.sol [hunk 1]
```diff
@@ -184,17 +177,7 @@ contract SetToken is
         // Check that function caller is Core
         require(msg.sender == ISetFactory(factory).core(), "ONLY_CORE_CAN_BURN_SET");
 
-        // Require user has tokens to burn
```
Comment (hydrazines) [inline contracts/core/SetToken.sol]: We still want this check right?

Comment (felix2feng) [reply contracts/core/SetToken.sol]: It's actually checked already in the `_burn` function of ERC20.sol

File: contracts/core/RebalancingSetToken.sol [hunk 2]
```diff
@@ -532,16 +522,9 @@ contract RebalancingSetToken is
...
 
         // Require user has tokens to burn
-        require(balances[_from] >= _quantity, "NOT_ENOUGH_TOKENS_TO_BURN");
-
-        // Update token balance of user
-        balances[_from] = balances[_from].sub(_quantity);
-
-        // Update total supply of Set Token
-        totalSupply_ = totalSupply_.sub(_quantity);
+        require(balanceOf(_from) >= _quantity, "NOT_ENOUGH_TOKENS_TO_BURN");
```
Comment (hydrazines) [inline contracts/core/RebalancingSetToken.sol]: Should we get rid of this check here then?

File: contracts/core/SetToken.sol (hunk 1 above)
Comment (hydrazines) [reply contracts/core/SetToken.sol]: Oh gotcha, see my other comment

File: contracts/core/RebalancingSetToken.sol (hunk 2 above)
Comment (felix2feng) [reply contracts/core/RebalancingSetToken.sol]: Yes, we can do that :)

**Comment by justinkchen** [comment]
Need to update `.solcover.js` too i think

File: contracts/core/RebalancingSetToken.sol [hunk 3]
```diff
@@ -492,24 +492,14 @@ contract RebalancingSetToken is
         (managerFee, protocolFee) = calculateFeeSplit(totalFees);
 
         // Update token balance of the manager
-        balances[_issuer] = balances[_issuer].add(issuerTotal);
-        balances[manager] = balances[manager].add(managerFee);
+        _mint(_issuer, issuerTotal);
```
Comment (asoong) [inline contracts/core/RebalancingSetToken.sol]: That's very convenient

Comment (asoong) [reply contracts/core/RebalancingSetToken.sol]: @bweick

File: contracts/mocks/tokens/BadTokenMock.sol [hunk 4]
```diff
@@ -31,7 +30,7 @@ contract BadTokenMock is StandardToken {
   */
   function transfer(address _to, uint256 _value) public returns (bool) {
     require(_to != address(0));
-    require(_value <= balances[msg.sender]);
+    require(_value <= balanceOf(msg.sender));
```
Comment (asoong) [inline contracts/mocks/tokens/BadTokenMock.sol]: I wonder if this adds to the call stack by making it a function that just accesses the mapping

Comment (felix2feng) [reply contracts/mocks/tokens/BadTokenMock.sol]: I assume it does

**Pull request merged**